### PR TITLE
[WIP] Nvidia update

### PR DESCRIPTION
**Pull request by SoapGentoo**
@ionenwks

### CODE REVIEW

File: x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild [hunk 1]
```diff
@@ -1,44 +1,38 @@
...
 
 EAPI=7
+
 inherit desktop flag-o-matic linux-info linux-mod multilib-minimal \
 	nvidia-driver portability systemd toolchain-funcs unpacker udev
 
-AMD64_FBSD_NV_PACKAGE="NVIDIA-FreeBSD-x86_64-${PV}"
 AMD64_NV_PACKAGE="NVIDIA-Linux-x86_64-${PV}"
-ARM_NV_PACKAGE="NVIDIA-Linux-armv7l-gnueabihf-${PV}"
+ARM64_NV_PACKAGE="NVIDIA-Linux-aarch64-${PV}"
```
Comment (mgorny) [inline x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild:10]: I'm tad confused here. All ebuilds add `ARM64_NV_PACKAGE` but none of them actually uses it.

File: x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild [hunk 2]
```diff
@@ -48,46 +42,41 @@ COMMON="
 		x11-libs/libXv
 		x11-libs/libXxf86vm
 		x11-libs/pango[X]
+		gtk3? ( x11-libs/gtk+:3 )
```
Comment (mgorny) [inline x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild:45]: Why are we defaulting to GTK+2?

File: x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild [hunk 3]
```diff
@@ -153,8 +118,8 @@ src_configure() {
 
 src_prepare() {
 	local man_file
-	for man_file in "${NV_MAN}"/*1.gz; do
-		gunzip $man_file || die
+	for man_file in *1.gz; do
+		gunzip "${man_file}" || die
```
Comment (mgorny) [inline x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild:122]:
```suggestion
	gunzip *1.gz || die
```

File: x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild [hunk 4]
```diff
@@ -223,7 +186,7 @@ donvidia() {
 	# Get just the library name
 	nv_LIBNAME=$(basename "${nv_LIB}")
 
-	if [[ "${nv_DEST}" ]]; then
+	if [[ ${nv_DEST} ]]; then
```
Comment (mgorny) [inline x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild:189]:
```suggestion
	if [[ -n ${nv_DEST} ]]; then
```

I personally prefer to use -n/-z when it's really about null/non-null value vs pseudo-booleans.

File: x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild [hunk 5]
```diff
@@ -236,15 +199,15 @@ donvidia() {
 
 	# If the library has a SONAME and SONAME does not match the library name,
 	# then we need to create a symlink
-	if [[ ${nv_SOVER} ]] && ! [[ "${nv_SOVER}" = "${nv_LIBNAME}" ]]; then
+	if [[ ${nv_SOVER} ]] && ! [[ ${nv_SOVER} == ${nv_LIBNAME} ]]; then
```
Comment (mgorny) [inline x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild:202]: Go bash all the way?

```suggestion
	if [[ -n ${nv_SOVER} && ${nv_SOVER} != ${nv_LIBNAME} ]]; then
```

File: x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild [hunk 6]
```diff
@@ -266,50 +229,38 @@ src_install() {
...
-		donvidia ${NV_OBJ}/libnvidia-encode.so.${NV_SOVER}
-	fi
+	donvidia libnvcuvid.so.${PV}
+	donvidia libnvidia-encode.so.${PV}
 
 	if use X; then
 		# Xorg DDX driver
 		insinto /usr/$(get_libdir)/xorg/modules/drivers
-		doins ${NV_X11}/nvidia_drv.so
+		doins nvidia_drv.so
```
Comment (mgorny) [inline x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild:247]: exeinto/doexe?

File: x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild [hunk 7]
```diff
@@ -409,140 +351,105 @@ src_install() {
...
-				"libGLESv2.so.2.1.0 ${GL_ROOT}"
-				"libGLX.so.0 ${GL_ROOT}"
-				"libGLdispatch.so.0 ${GL_ROOT}"
-				"libOpenGL.so.0 ${GL_ROOT}"
-			)
-		fi
 
-		if use wayland && [[ ${ABI} == "amd64" ]];
-		then
+		if use wayland && [[ ${ABI} == "amd64" ]]; then
```
Comment (mgorny) [inline x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild:401]: As a next step, would be nice to finally switch to the new multilib eclasses.

File: x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild (hunk 1 above)
Comment (SoapGentoo) [reply x11-drivers/nvidia-drivers/nvidia-drivers-460.39.ebuild:10]: correct, this is just preparing everything for `arm64` support, which I'll add soon

**Pull request closed without merging**